### PR TITLE
Issue #338: Added onEvaluatedDataChange  prop to expose the result of the formula

### PR DESCRIPTION
**Pull request by BALURAM1226**
Issue #338: Added getEvaluatedData prop to expose the result of the formula

### CODE REVIEW

File: src/Spreadsheet.tsx [hunk 1]
```diff
@@ -119,6 +119,8 @@ export type Props<CellType extends Types.CellBase> = {
     nextCell: null | CellType,
     coords: null | Point.Point
   ) => void;
+  /** Callback called when the Spreadsheet's data changes and get realtime evaluated data. */
+  getEvaluatedData :  (data: Matrix.Matrix<CellType>) => void;
```
Comment (iddan) [inline src/Spreadsheet.tsx:123]:
```suggestion
  /** Callback called when the Spreadsheet's evaluated data changes. */
  onEvaluatedDataChange :  (data: Matrix.Matrix<CellType>) => void;
```

Comment (BALURAM1226) [reply src/Spreadsheet.tsx:123]: Thanks for suggestion , I will make changes and then push it again

**Comment by iddan** [comment]
Please see CI failure. You can also run it locally with `yarn ci`

**Comment by BALURAM1226** [comment]
I have checked it, and there is a Prettier issue there. However, when I run Prettier on my code, many things are changed. Should I push that code file?

**Comment by BALURAM1226** [comment]
I tried Prettier ESLint It worked. let's try to merge again

**Comment by BALURAM1226** [comment]
I tried Prettier ESLint It worked. let's try to merge again

**Comment by BALURAM1226** [comment]
@iddan I have committed code with Prettier that resolved a CI failure issue. Can you please review the code?

**Comment by BALURAM1226** [comment]
@iddan , I have sent a review request for changes to address the resolved CI failure. Please take a moment to review.Thank you

**Comment by iddan** [comment]
Thank you, seems like CI is still failing

**Comment by BALURAM1226** [comment]
Screenshot 2024-02-22 210043
Screenshot 2024-02-22 211147


@iddan, I ran **npm run check-format** in my local code and identified code style issues in 96 files. After running **npm run format**, I resolved the code style issues in all 96 files. However, I only committed the changes to the **Spreadsheet.tsx** file , because in CI failure only in Spreadsheet.tsx file code style issue was found. I believe this should resolve the issue. If you have any other suggestions, please let me know

**Comment by BALURAM1226** [comment]
Screenshot 2024-02-29 201049

@iddan After running the 'npm run ci' command locally, I encountered an error. After fixing the above CI failure ( Property 'onEvaluatedDataChange' is missing in type ), should I push the commit that resolved the CI failure? Can you please provide suggestions regarding this below error? May this below error affect to CI ?

Error: No files matching the pattern "'*/**.css'" were found.
    at C:\New folder\react-spreadsheet\node_modules\stylelint\lib\standalone.js:212:12
ERROR: "lint-css" exited with 1.
ERROR: "lint" exited with 1.

**Comment by iddan** [comment]
I see TypeScript errors:
Screen Shot 2024-03-02 at 18 34 33

**Comment by BALURAM1226** [comment]
@iddan Yes, you are correct. I resolved the code issue in my local system. After that, I ran 'npm run CI' locally, and I encountered an error, as you can see in the screenshot above. I am seeking confirmation from you, and if the above error does not affect the CI failure, then I will commit the new code. I've been facing CI failures repeatedly, so if the mentioned error doesn't impact the CI failure, I'll proceed to commit the new code and try again for a successful merge

**Comment by iddan** [comment]
You can push, and if the CI command passes in GitHub Actions it's ok

**Comment by BALURAM1226** [comment]
@iddan , You need to publish new version for this changes for available for all npm package users ? Because added new props visible in storybook and that working but not show in npm package after installation

**Comment by BALURAM1226** [comment]
@iddan  Do you need to release a new version to make these changes available to all npm package users? Because added new props visible in storybook and that working but not show in npm package after installation

**Pull request merged**